packages/chopsticks/src/server.ts

```diff
@@ -9,7 +9,7 @@ const httpLogger = defaultLogger.child({ name: 'http' })
 const wsLogger = defaultLogger.child({ name: 'ws' })
 
 const singleRequest = z.object({
-  id: z.number(),
+  id: z.optional(z.union([z.number().int(), z.string(), z.null()])),
```

Comment:
why does this need to be null and undefined? don't we need either number or string?

Reply:
For polkadot-api we would only need it to accept strings, but I chose to make it spec-compliant instead. The spec for the id field on a request says:

> An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null and Numbers SHOULD NOT contain fractional parts
> [...]
> A Notification is a Request object without an "id" member. A Request object that is a Notification signifies the Client's lack of interest in the corresponding Response object, and as such no Response object needs to be returned to the client. The Server MUST NOT reply to a Notification, including those that are within a batch request.

Meaning undefined (not included) and null have different meaning: Not included means the request is a notification, which means it's not expecting any response back, whereas having a `null` id requires the server to reply with to the message (with the same null value for the id)

If you rather have a smaller change even if it's not spec-compliant I'm ok simplifying it back, but I thought this would have more value.